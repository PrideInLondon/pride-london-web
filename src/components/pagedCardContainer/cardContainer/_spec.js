import React from 'react'
import { shallow } from 'enzyme'
import CardContainer from '.'

describe('CardContainer', () => {
  it('renders component CardContainer', () => {
    // eslint-disable-next-line react/prop-types
    const CardComponent = ({ text }) => <div>{text}</div>

    const cardContent = [
      {
        id: 'id-foo',
        text: 'text-foo',
      },
      {
        id: 'id-bar',
        text: 'text-bar',
      },
      {
        id: 'id-baz',
        text: 'text-baz',
      },
    ]

    const wrapper = shallow(
      <CardContainer cardContent={cardContent} CardComponent={CardComponent} />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
