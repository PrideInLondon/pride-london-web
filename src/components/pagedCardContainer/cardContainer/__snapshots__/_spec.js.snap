// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer renders component CardContainer 1`] = `
<grid__Container
  className=""
  mx="auto"
>
  <Row
    flexWrap={null}
    ml={null}
    mr={null}
    mx={null}
  >
    <grid__FlexColumn
      key="id-foo"
      py={
        Array [
          2,
          2,
          2,
          3,
        ]
      }
      width={
        Array [
          1,
          1,
          0.5,
          0.3333333333333333,
        ]
      }
    >
      <CardComponent
        text="text-foo"
      />
    </grid__FlexColumn>
    <grid__FlexColumn
      key="id-bar"
      py={
        Array [
          2,
          2,
          2,
          3,
        ]
      }
      width={
        Array [
          1,
          1,
          0.5,
          0.3333333333333333,
        ]
      }
    >
      <CardComponent
        text="text-bar"
      />
    </grid__FlexColumn>
    <grid__FlexColumn
      key="id-baz"
      py={
        Array [
          2,
          2,
          2,
          3,
        ]
      }
      width={
        Array [
          1,
          1,
          0.5,
          0.3333333333333333,
        ]
      }
    >
      <CardComponent
        text="text-baz"
      />
    </grid__FlexColumn>
  </Row>
</grid__Container>
`;
